### ISSUE TITLE
tests: `pkg.sensors.tracing.TestKprobeRateLimit` is flaky

### ISSUE THREAD

**Issue by mtardy** [cilium/tetragon#2789]
https://github.com/cilium/tetragon/actions/runs/10316714693/job/28559512984?pr=2787

[tetragon-bugtool.zip](https://github.com/user-attachments/files/16562376/tetragon-bugtool.zip)
[tetragon-json.zip](https://github.com/user-attachments/files/16562378/tetragon-json.zip)

**Comment by calghar** [comment]
@mtardy I also saw this issue fail on a recent test I ran but on Windows. I believe its a race condition that pops up occasionally. Basically because of the sleep time(2s) before the main check, sometimes an extra event can be emitted resulting in this test failing. Does this make sense? Or is my assessment off here? 
A simple solution could be to just have a polling system that checks for the number of events emitted and exits when the required number of events is reached. its 10 atm iirc.

**Comment by mtardy** [comment]
> [@mtardy](https://github.com/mtardy) I also saw this issue fail on a recent test I ran but on Windows. I believe its a race condition that pops up occasionally. Basically because of the sleep time(2s) before the main check, sometimes an extra event can be emitted resulting in this test failing. Does this make sense? Or is my assessment off here? A simple solution could be to just have a polling system that checks for the number of events emitted and exits when the required number of events is reached. its 10 atm iirc.

Yeah I haven't looked into this but that could make sense. If you find a solution to make this more reliable (and not too slow) that would be great :)

**Comment by calghar** [comment]
You can assign this to me @mtardy. Will try to sort this out asap.
